### PR TITLE
En 10239/transfer esdt tokens

### PR DESCRIPTION
**Pull request by mariusmihaic**
- Added dependency to GNU GMP library to handle big int 
- Added `BigUInt` class
- Added `SCArguments` class to handle smart contract arguments call for transactions
- Added function to prepare ESDT token transfer with or without SC function & parameters
- Added gtests

### CODE REVIEW

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

File: include/internal/biguint.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#ifndef ERD_BIG_UINT_H
+#define ERD_BIG_UINT_H
+
+#include <string>
+
+class BigUInt
+{
+public:
+    explicit BigUInt(std::string value);
```
Comment (andreibancioiu) [inline include/internal/biguint.h:9]: 👍

File: include/transaction/scarguments.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#include "internal/biguint.h"
+#include "account/address.h"
+
+class SCArguments
+{
+public:
+    explicit SCArguments();
+
+    void add(std::string const &arg);
```
Comment (andreibancioiu) [inline include/transaction/scarguments.h:14]: 👍

File: src/transaction/scarguments.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    m_args.emplace_back(val);
+}
+
+void SCArguments::add(Address const &arg)
+{
+    bytes const pk = arg.getPublicKey();
+
+    std::string pkHex(pk.begin(), pk.end());
+    pkHex = util::stringToHex(pkHex);
```
Comment (andreibancioiu) [inline src/transaction/scarguments.cpp:27]: Perhaps in the future we can add `toHex` (`toPubKey`) to `Address` class as well.

File: src/transaction/scarguments.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    return m_args.empty();
+}
+
+std::string SCArguments::asOnData() const
+{
+    std::string ret;
+
+    for (auto const &arg : m_args)
+    {
+        ret += "@" + arg;
```
Comment (andreibancioiu) [inline src/transaction/scarguments.cpp:43]: So, `asOnData()` always returns having a leading `@`, right?

File: tests/test_src/test_transaction/test_transaction.cpp [hunk 5]
```diff
@@ -495,3 +496,210 @@ TEST_F(TransactionSerializeFixture, serialize_missingFields)
     tx.m_receiver = nullptr;
     expectSerializeException<std::invalid_argument>(tx, ERROR_MSG_RECEIVER);
 }
+
+TEST(SCArguments, add_emtpy_asOnData)
```
Comment (andreibancioiu) [inline tests/test_src/test_transaction/test_transaction.cpp:500]: 👍

**Review by andreibancioiu — APPROVED** [review]

File: include/internal/biguint.h (hunk 1 above)
Comment (mariusmihaic) [reply include/internal/biguint.h:9]: Thanks ! :+1:

File: include/transaction/scarguments.h (hunk 2 above)
Comment (mariusmihaic) [reply include/transaction/scarguments.h:14]: Thanks ! :+1:

File: tests/test_src/test_transaction/test_transaction.cpp (hunk 5 above)
Comment (mariusmihaic) [reply tests/test_src/test_transaction/test_transaction.cpp:500]: Thanks ! :+1:

File: src/transaction/scarguments.cpp (hunk 4 above)
Comment (mariusmihaic) [reply src/transaction/scarguments.cpp:43]: Yes, it will always return the params preceded with `@`

File: src/transaction/scarguments.cpp (hunk 3 above)
Comment (mariusmihaic) [reply src/transaction/scarguments.cpp:27]: Maybe we could even:

- Change bytes typedef to be a class
- Add `toHex` method to it ?

``` c++
class Bytes(std::string input)
public: 
std::string toHex();
```

**Pull request merged**